src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -51,14 +51,15 @@ auto evaluate(const T& rhs_te) {
   using rhs = tmpl::transform<tmpl::remove_duplicates<typename T::args_list>,
                               std::decay<tmpl::_1>>;
   static_assert(
-      tmpl::equal_members<tmpl::list<std::decay_t<LhsTensorIndices>...>,
-                          rhs>::value,
+      tmpl::equal_members<
+          tmpl::list<std::decay_t<decltype(LhsTensorIndices)>...>, rhs>::value,
       "All indices on the LHS of a Tensor Expression (that is, those specified "
       "in evaluate<Indices::...>) must be present on the RHS of the expression "
       "as well.");
 
   using rhs_tensorindex_list = typename T::args_list;
-  using lhs_tensorindex_list = tmpl::list<LhsTensorIndices...>;
+  using lhs_tensorindex_list =
+      tmpl::list<std::remove_reference_t<decltype(LhsTensorIndices)>...>;
```

Comment:
Does `decay_t` not work here? I worry that someone will be confused that you use `decay_t` above and `remove_reference_t` here

Reply:
Updated! It seems that either would work for the purpose we need here